### DISCUSSION TITLE
Are there any plans for a nicer grammar->parser library interface?

### DISCUSSION THREAD

**Discussion by alexmozaidze** [tree-sitter/tree-sitter#2864] [Q&A] [answered]
From what I've gathered, the only crate capable of generating a parser from a grammar is `tree-sitter-cli`, but its interface is rather clunky:
- It heavily makes use of `anyhow::Result`
- Uses `&PathBuf` instead of `&Path`
- [`generate_parser_in_directory`](https://docs.rs/tree-sitter-cli/0.20.8/tree_sitter_cli/generate/fn.generate_parser_in_directory.html) is the only public function to actually generate a grammar (a direct grammar to parser function would be nice)

Are there any plans to move it into a separate library/crate and refine it?

**Comment by amaanq** [comment, marked as answer]
The generate crate is now its own library, and has typed errors in a decent amount of spots, so I guess this can be considered solved

**Reply by alexmozaidze** [reply]
Poggers
